WLEmptyState/Classes/Extensions/UITableView+WLEmptyState.swift

```diff
@@ -44,26 +44,44 @@ extension UITableView: WLEmptyStateProtocol {
         }
     }
     
+    /// The original value before enable/disable scrolling.
+    var originalScrollingValue: Bool {
+        get {
+            return (objc_getAssociatedObject(self, &AssociatedKeys.originalScrollingValue) as? Bool) ?? isScrollEnabled
+        }
+        
+        set {
+            objc_setAssociatedObject(self, &AssociatedKeys.originalScrollingValue, newValue, .OBJC_ASSOCIATION_COPY)
+        }
+    }
+    
     @objc private dynamic func swizzledReload() {
-        guard emptyStateDataSource != nil else { return }
-
         swizzledReload()
         
+        guard emptyStateDataSource != nil else { return }
+        
         if numberOfItems == 0 && self.subviews.count > 1 {
+            originalScrollingValue = isScrollEnabled
+            isScrollEnabled = emptyStateDelegate?.emptyStateShouldScroll() ?? true
+            
             addSubview(emptyStateView)
             if let emptyStateView = emptyStateView as? EmptyStateView {
-                emptyStateView.titleLabel.attributedText = self.emptyStateDataSource?.titleForEmptyDataSet()
-                emptyStateView.descriptionLabel.attributedText = self.emptyStateDataSource?.descriptionForEmptyDataSet()
-                emptyStateView.image.image = self.emptyStateDataSource?.imageForEmptyDataSet()
+                let datasource = self.emptyStateDataSource
+                emptyStateView.titleLabel.attributedText = datasource?.titleForEmptyDataSet()
+                emptyStateView.descriptionLabel.attributedText = datasource?.descriptionForEmptyDataSet()
+                emptyStateView.image.image = datasource?.imageForEmptyDataSet()
```

Comment:
Can we rename the image from `emptyStateView ` to `imageView` to reflect its identity? 🙏